x-pack/plugins/cases/server/client/cases/bulk_create.ts

```diff
@@ -0,0 +1,191 @@
+import { CaseSeverity, UserActionTypes } from '../../../common/types/domain';
+import { decodeWithExcessOrThrow } from '../../../common/api';
+
+import { Operations } from '../../authorization';
+import { createCaseError } from '../../common/error';
+import { flattenCaseSavedObject, isSOError, transformNewCase } from '../../common/utils';
+import type { CasesClient, CasesClientArgs } from '..';
+import { LICENSING_CASE_ASSIGNMENT_FEATURE } from '../../common/constants';
+import { decodeOrThrow } from '../../../common/api/runtime_types';
+import type {
+  BulkCreateCasesRequest,
+  BulkCreateCasesResponse,
+  CasePostRequest,
+} from '../../../common/types/api';
+import { BulkCreateCasesResponseRt, BulkCreateCasesRequestRt } from '../../../common/types/api';
+import {} from '../utils';
+import { validateCustomFields } from './validators';
+import { fillMissingCustomFields } from './utils';
+import type { BulkCreateCasesArgs } from '../../services/cases/types';
+import type { NotifyAssigneesArgs } from '../../services/notifications/types';
+
+/**
+ * Bulk create new case.
+ *
+ */
+export const bulkCreate = async (
+  data: BulkCreateCasesRequest,
+  clientArgs: CasesClientArgs,
+  casesClient: CasesClient
+): Promise<BulkCreateCasesResponse> => {
+  const {
+    services: { caseService, userActionService, licensingService, notificationService },
+    user,
+    logger,
+    authorization: auth,
+  } = clientArgs;
+
+  try {
+    const decodedData = decodeWithExcessOrThrow(BulkCreateCasesRequestRt)(data);
+    const configurations = await casesClient.configure.get();
+
+    const customFieldsConfigurationMap: Map<string, CustomFieldsConfiguration> = new Map(
+      configurations.map((conf) => [conf.owner, conf.customFields])
+    );
+
+    const casesWithIds = decodedData.cases.map((theCase) => ({
+      ...theCase,
+      id: theCase.id ?? SavedObjectsUtils.generateId(),
+    }));
+
+    await auth.ensureAuthorized({
+      operation: Operations.createCase,
+      entities: casesWithIds.map((theCase) => ({ owner: theCase.owner, id: theCase.id })),
+    });
+
+    const hasPlatinumLicenseOrGreater = await licensingService.isAtLeastPlatinum();
+
+    const bulkCreateRequest: BulkCreateCasesArgs['cases'] = [];
+
+    for (const theCase of casesWithIds) {
+      const customFieldsConfiguration = customFieldsConfigurationMap.get(theCase.owner);
+
+      const customFieldsValidationParams = {
+        requestCustomFields: theCase.customFields,
+        customFieldsConfiguration,
+      };
+
+      validateCustomFields(customFieldsValidationParams);
+
+      /**
+       * Assign users to a case is only available to Platinum+
+       */
+
+      if (theCase.assignees && theCase.assignees.length !== 0) {
+        if (!hasPlatinumLicenseOrGreater) {
+          throw Boom.forbidden(
+            'In order to assign users to cases, you must be subscribed to an Elastic Platinum license'
```

Comment:
Should we mention the specific case ID here?

Reply:
> The reason is that on a create request most of the time the ID will be automatically generated automatically and it will not add any value.

Didn't think about this, good point.